code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm

```diff
@@ -806,19 +849,29 @@
 	color = "#C8A5DC"
 	metabolization_rate = 0.5 * REAGENTS_METABOLISM
 	overdose_threshold = 60
+	stun_threshold = 4
+	speedboost = FAST
 
 /datum/reagent/medicine/stimulants/on_mob_life(mob/living/M)
-	M.status_flags |= GOTTAGOFAST
 	if(M.health < 50 && M.health > 0)
 		M.adjustOxyLoss(-1*REM)
 		M.adjustToxLoss(-1*REM)
 		M.adjustBruteLoss(-1*REM)
 		M.adjustBloodLoss(-0.1*REM)
 		M.adjustFireLoss(-1*REM)
-	M.AdjustParalysis(-3)
-	M.AdjustStunned(-3)
-	M.AdjustWeakened(-3)
-	M.adjustStaminaLoss(-5*REM)
+	if(!(M.stunned || M.weakened || M.paralysis))
+		stun_timer++
+		metabolization_rate = 0.5* REAGENTS_METABOLISM
```

Comment:
Could you not use initial(metabolization_rate) is your multiplier?
So if someone wants to change the metabolizationrate, they only need to change the value at the definitions of the reagent?